### PR TITLE
Allow different environment configs in now.json and package.json

### PR DESCRIPTION
**Pull request by mike-engel**
**tl;dr: Users can now, optionally, customize now.json & package.json per environment**

This should provide a nice, backwards compatible solution to #275. This refers to the value of `NODE_ENV` when set to dynamically use a set of now config options per environment. If `NODE_ENV` is set, but no environments are established, it does the existing behavior which is to assume the keys in the root of the json object are the config.

### Examples
The backwards compatible way
```json
{
  "alias": "foo.com"
}
```

The optional new way
```json
{
  "staging": {
    "alias": "dev.foo.com"
  },
  "production": {
    "alias": "foo.com"
  }
}
```

### Changes
- adds support for environment specific now configs
- adds `prettier` to the dev dependencies since it runs when committing and wasn't being installed
- adds a test for existing functionality, which is when no environment is configured

### Caveats
- If they have environments configured in `now.json` or `package.json` and don't specify `NODE_ENV, then it won't deploy correctly

### Doc updates
If this change is accepted, the docs will need to be updated. Obviously I can't write them, but here is a rough draft:

<details>
<summary><strong>rough draft</strong></summary>
All of the properties mentioned below can be used both in the package.json and inside the now.json file. Additionally, if you need support for multiple environments, each property can be specified in separate environment objects and used by setting the `NODE_ENV` environment variable:

```
{
  "staging": {
    "name": "app-staging",
    "alias": "dev.app.com"
  },
  "production": {
    "name": "app",
    "alias": "app.com"
  }
}
```
</details>

### Other considerations
I did consider using different files per environment (i.e. `now.staging.json`, `now.production.json`, etc. I decided against that, however, as there could be tons of files which isn't so great. Additionally, having everything in a single file reduces the cognitive load and keeps everything in one place.

### CODE REVIEW

**Comment by leo** [comment]
Thank you for your suggestion and the effort you've put into this, @mike-engel! I would be very happy to merge another PR from you in the future.

However, please make sure to discuss the idea with us and let someone from the team confirm it, before you put that much effort into a PR again. Because in this case, the solution will be another flag, which would allow devs to specify the location of the file: https://github.com/zeit/now-cli/issues/275

🙏 😊

**Comment by mike-engel** [comment]
Thanks Leo for the feedback. Normally I agree with you, and given a large amount of work for a PR, I would much prefer to discuss it before hand (like #308).

Luckily this PR was incredibly small and didn't take much effort. It seems like you all are pretty busy lately, so I figured throwing up a PR, whether it was wrong or not, would be better than bike shedding.

No hard feelings, and thanks for helping to make ZEIT and now awesome 😊

**Comment by leo** [comment]
@mike-engel ❤️
